src/tsomi-rdf.js

```diff
@@ -313,37 +247,6 @@ function binding_to_string(binding) {
   return result;
 }
 
-function sparqlQuery(query, variables, callback) {
-
-  var execute = function() {
-    Object.keys(variables).forEach(function(variable) {
-      query = query.replace(new RegExp('%' + variable + '%', 'g'), variables[variable]);
-    });
-
-    query = prefix_table_to_string(prefixies) + '\n' + query;
-
-    if (debugging) {
-      console.log('---------------- query ----------------');
-      console.log(query);
-      console.log('^^^^^^^^^^^^^^^^ query ^^^^^^^^^^^^^^^^');
-    }
-
-    $.getJSON(QUERY_URL + escape(query))
-      .then(function(data) {
-        if (debugging) {
-          console.log('---------------- results -----------------');
-          display_results(data);
-          console.log('^^^^^^^^^^^^^^^^ results ^^^^^^^^^^^^^^^^');
-        }
-
-        callback(data);
-      }, function(error) {
-        console.log('HTTP error'), callback(undefined);
-      });
-  };
-
-  setTimeout(execute, 0);
```

Comment:
What was your thinking when dropping the `setTimeout` wrapper?  I suspect that this was put here because things were hanging at some point.

Reply:
i suspect that was only there as a way to hack around a race condition that was happening somewhere. i can't replicate it, so i don't think it's still a problem?